runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -2366,71 +2366,82 @@ void jitClassesRedefined(J9VMThread * currentThread, UDATA classCount, J9JITRede
 
    fe->releaseCompilationLock();
 
-   classPair = classList;
-   for (i = 0; i < classCount; i++)
+   if (!extensionsUsed)
```

Comment:
Unless I am missing something, I think we should remove this `if` statement. Because there is no class communicated by the VM, the `for` loop that follows will not be executed. Also, if in the future the VM decides to send us classes in the extended case as well, we should not avoid processing those.